bwscanner/attacher.py

```diff
@@ -156,30 +156,33 @@ def connect_to_tor(launch_tor, circuit_build_timeout, control_port=None,
     if tor_overrides:
         tor_options.update(tor_overrides)
 
+    tor_config = txtorcon.TorConfig()
+
+    # Update Tor config options from dictionary
+    for key, value in tor_options.items():
+        setattr(tor_config, key, value)
+
     if launch_tor:
         log.info("Spawning a new Tor instance.")
         # TODO: Pass in data_dir directory so consensus can be cached
-        tor = yield txtorcon.launch(reactor)
+        # Launch tor with config, in order to don't get CONF_CHANGED when
+        # updating options that can't be changed while tor is running.
+        tor = yield txtorcon.launch(reactor, _tor_config=tor_config)
     else:
```

Comment:
When launching tor like this (and passing in the config) you shouldn't have to assign the object nor do `.save()` (as below).

Reply:
Just to clarify, ``.save()`` is not being call when launching tor, in this part of the ``if``, it's call when connecting to ``tor`` in the ``else`` below, as you point out.